fdbserver/workloads/BackupToDBCorrectness.actor.cpp

```diff
@@ -667,10 +667,38 @@ struct BackupToDBCorrectnessWorkload : TestWorkload {
 				// wait(diffRanges(self->backupRanges, self->backupPrefix, cx, self->extraDB));
 
 				state Standalone<VectorRef<KeyRangeRef>> restoreRange;
+				state bool containsSystemKeys = false;
 				for (auto r : self->backupRanges) {
-					restoreRange.push_back_deep(
-					    restoreRange.arena(),
-					    KeyRangeRef(r.begin.withPrefix(self->backupPrefix), r.end.withPrefix(self->backupPrefix)));
+					if (!r.intersects(getSystemBackupRanges())) {
+						restoreRange.push_back_deep(
+						    restoreRange.arena(),
+						    KeyRangeRef(r.begin.withPrefix(self->backupPrefix), r.end.withPrefix(self->backupPrefix)));
+					} else {
+						containsSystemKeys = true;
+					}
+				}
+
+				// restore system keys first before restoring user data
```

Comment:
I don't know if we want to do this unconditionally, since the restore of everything should work fine if encryption is not enabled. Instead, we could do it only if encryption is enabled (and perhaps randomly if it isn't).

Also, this is essentially asserting that if we are restoring system keys, then we are intending to restore the entire set of system backup ranges. It's possible that isn't true, and maybe we want to account for that somehow as well.

Reply:
Earlier today in our morning syncup meeting and I had similar feedback for him , plan is to update code to enforce ordering **iff** encryption is enabled.